### PR TITLE
nestloop optimize  inner 1472

### PR DESCRIPTION
**Pull request by ylinzhu**
Reason:  
 Improve 
Type:  
  Improve  
Influences：  
   fix xx

### CODE REVIEW

**Comment by ylinzhu** [comment]
parameters need to be replace

File: src/main/java/com/actiontech/dble/plan/optimizer/MyOptimizer.java [hunk 1]
```diff
@@ -75,10 +74,10 @@ public static PlanNode optimize(PlanNode node, @Nullable HintPlanInfo hintPlanIn
                 LimitPusher.optimize(node);
 
                 node = SelectedProcessor.optimize(node);
-
-                boolean useJoinStrategy = SystemConfig.getInstance().isUseJoinStrategy();
-                if (useJoinStrategy) {
-                    JoinStrategyProcessor.optimize(node);
+                if (!hintPlanInfo.isEmpty()) {
+                    HintStrategyProcessor.optimize(node, hintPlanInfo);
```
Comment (dcy10000) [inline src/main/java/com/actiontech/dble/plan/optimizer/MyOptimizer.java:78]: The naming is inappropriate

**Review by dcy10000 — CHANGES_REQUESTED** [review]

File: src/main/java/com/actiontech/dble/plan/optimizer/JoinNestLoopChooser.java [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    public void tryNestLoop() throws MySQLOutPutException {
+        buildNodeMap(jn);
+        buildHintDependency();
+        traverseNode(jn);
+    }
+
+    private void buildNodeMap(JoinNode joinNode) {
+        PlanNode leftNode = joinNode.getLeftNode();
+        PlanNode rightNode = joinNode.getRightNode();
+        if (leftNode instanceof JoinNode) {
```
Comment (dcy10000) [inline src/main/java/com/actiontech/dble/plan/optimizer/JoinNestLoopChooser.java:38]: right join maybe JoinNode too.
like  /*!dble:plan=a&b&c*/select * from a right join b on a.id =b.id right join c on b.id=c.id ;

**Review by dcy10000 — CHANGES_REQUESTED** [review]

File: src/main/java/com/actiontech/dble/plan/optimizer/JoinNestLoopChooser.java [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+    private PlanNode findDependNode(PlanNode node) {
+        JoinNode joinNode = (JoinNode) node.getParent();
+        String firstTableName = null;
+        List<ItemFuncEqual> joinFilter = joinNode.getJoinFilter();
+        for (ItemFuncEqual itemFuncEqual : joinFilter) {
+            List<Item> arguments = itemFuncEqual.arguments();
+            String tableName = arguments.get(0).getTableName();
+            firstTableName = Optional.ofNullable(firstTableName).orElse(tableName);
+            HintPlanNodeGroup group = hintDependMap.get(node.getAlias());
+            if (group.getType() == HintPlanNodeGroup.Type.ER) {
```
Comment (dcy10000) [inline src/main/java/com/actiontech/dble/plan/optimizer/JoinNestLoopChooser.java:104]: this compare doesn't anything about itemFuncEqual?

File: src/main/java/com/actiontech/dble/plan/optimizer/JoinNestLoopChooser.java [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+                return nodeMap.get(tableName).getParent();
+            }
+            if (!nodeMap.containsKey(tableName)) {
+                return nodeMap.get(tableName);
```
Comment (dcy10000) [inline src/main/java/com/actiontech/dble/plan/optimizer/JoinNestLoopChooser.java:108]: always return  NULL ?

**Review by dcy10000 — CHANGES_REQUESTED** [review]

**Review by dcy10000 — APPROVED** [review]
lgtm

**Pull request merged**